rfcs/20190509-keras-saved-model.md

```diff
@@ -0,0 +1,226 @@
+# Keras SavedModel saving/loading 
+
+| Status        | Proposed                                             |
+:-------------- |:---------------------------------------------------- |
+| **Author(s)** | Kathy Wu (kathywu@google.com)                        |
+| **Sponsor**   | Karmel (karmel@google.com)                           |
+| **Updated**   | 2019-05-09                                           |
+
+## Objective
+
+Add ability to (1) save a 
+[TensorFlow SavedModel](https://www.tensorflow.org/alpha/guide/saved_model) 
+from a Keras model and (2) load a Keras model from a SavedModel. This is a 
+standalone serialization format, which allows models with custom objects to 
+be saved/loaded without having the original code (currently required by the HDF5 and JSON formats). 
+
+**Goals:**
+
+- Add SavedModel format for saving Keras models. 
+- Support comprehensive model serialization/deserialization 
+  - Define model serialization coverage (what parts of the model are saved 
+    to the SavedModel) 
+  - Address serialization pattern that is reusable across different
+    frameworks (TensorFlow Hub, tf.module, etc.) 
+  - Define SavedModel deserialization (properties of the reconstructed 
+    model) 
+
+**Non-goals:**
+
+- Serializing custom training loops. Only the standard `model.compile` to 
```

Comment:
To confirm, custom training loops will not be serialized by default, but it will still be possible to load a keras saved model and write a custom training loop for it, right?

Also we should probably say that if you want to serialize your custom training loop you can wrap it in tf.function and export that function as part of the savedmodel, right?

Reply:
Yes, the loaded keras model will behave identically to the original model. And yeah, attaching the training loop to the model will work. Added a comment to the doc.